module/product/src/Domain/Entity/AbstractProduct.php

```diff
@@ -201,8 +201,10 @@ public function changeAttributes(array $attributes): void
                 $this->addAttribute($attributeCode, $attribute);
             }
         }
+        $systemAttributes = preg_grep('/^esa_[\d]*/', array_keys($this->attributes));
```

Comment:
I think, if no better solution we should delimit it to specific attributes - `esa_created_at` etc...